### PR TITLE
Adding a flag to able/disable scope info

### PR DESCRIPTION
**Pull request by BrianBohe**
We are adding a [new way of tracking variables' home](https://github.com/dotnet/coreclr/pull/22770). It is useless to have both activated at the same time, so this PR is for able/disable siScope and psiScope info.

### CODE REVIEW

**Comment by BrianBohe** [comment]
@CarolEidt @BruceForstall @sandreenko @mikedn

**Comment by BrianBohe** [comment]
After all tests have run, I will commit changing flag to be 1 by default.

**Comment by sandreenko** [comment]
@dotnet-bot test Windows_NT x64 Checked pmi_asm_diffs
@dotnet-bot test Windows_NT x86 Checked pmi_asm_diffs

**Comment by BrianBohe** [comment]
I have tested this building this branch on debug mode for x86 and moving the built dlls and pdbs to a peace of code I know its result. I got 0 bytes allocated to siSope and finished correctly.

File: src/jit/codegenlinear.cpp [hunk 1]
```diff
@@ -350,9 +351,9 @@ void CodeGen::genCodeForBBlist()
         /* Tell everyone which basic block we're working on */
 
         compiler->compCurBB = block;
-
+#ifdef USUSING_SCOPE_INFOING_
```
Comment (jkotas) [inline src/jit/codegenlinear.cpp]: Typo?

Comment (BrianBohe) [reply src/jit/codegenlinear.cpp]: Oh, thanks!

Comment (BrianBohe) [reply src/jit/codegenlinear.cpp]: Oh, thanks!

**Comment by BrianBohe** [comment]
I have rebased master

**Comment by BrianBohe** [comment]
> 
> 
> @dotnet-bot test Windows_NT x64 Checked pmi_asm_diffs
> @dotnet-bot test Windows_NT x86 Checked pmi_asm_diffs

What should I see with those commands? @sandreenko

**Comment by sandreenko** [comment]
> What should I see with those commands? @sandreenko

These are the jobs that run jit-diff to compare asm output from baseline CoreCLR with your changes.
So it is useful to trigger them to conform that asm diffs are as expected.
For your PR [the results](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_windows_nt_pmi_asm_diffs_prtest/24/console
) are:
```
11:27:16 Summary:
11:27:16 (Lower is better)
11:27:16 
11:27:16 Total bytes of diff: 0 (0.00 % of base)
11:27:16 
11:27:16 0 total files with size differences (0 improved, 0 regressed), 129 unchanged.
11:27:16 
11:27:16 0 total methods with size differences (0 improved, 0 regressed), 246983 unchanged.
```
so no diffs as expected.

[More information about jit-diff.](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_windows_nt_pmi_asm_diffs_prtest/24/console
)

File: src/jit/codegeninterface.h [hunk 2]
```diff
@@ -25,6 +25,12 @@
 #include "jitgcinfo.h"
 #include "treelifeupdater.h"
 
+// Disable this flag to avoid using psiScope/siScope info to report reporting
```
[the remark this answers is not in the document]
Comment (BrianBohe) [reply src/jit/codegeninterface.h]: Yes, I commit changes which disable ScopeInfo just to see the results of tests on other architectures. But I will leave it defined as default before this pr is merged.

File: src/jit/codegen.h [hunk 3]
```diff
@@ -561,7 +561,11 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                          siVarLoc*      varLoc);
 
     void genSetScopeInfo();
+#ifdef USING_SCOPE_INFO
+    void genSetScopeInfoUsingsiScope();
+#endif
```
[the remark this answers is not in the document]
Comment (BrianBohe) [reply src/jit/codegen.h]: > 
> 
> > What should I see with those commands? @sandreenko
> 
> These are the jobs that run jit-diff to compare asm output from baseline CoreCLR with your changes.
> So it is useful to trigger them to conform that asm diffs are as expected.
> For your PR [the results](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_windows_nt_pmi_asm_diffs_prtest/24/console) are:
> 
> ```
> 11:27:16 Summary:
> 11:27:16 (Lower is better)
> 11:27:16 
> 11:27:16 Total bytes of diff: 0 (0.00 % of base)
> 11:27:16 
> 11:27:16 0 total files with size differences (0 improved, 0 regressed), 129 unchanged.
> 11:27:16 
> 11:27:16 0 total methods with size differences (0 improved, 0 regressed), 246983 unchanged.
> ```
> 
> so no diffs as expected.
> 
> [More information about jit-diff.](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_windows_nt_pmi_asm_diffs_prtest/24/console)

The second link is the same as the first one. Did you mean that ? @sandreenko

**Comment by BrianBohe** [comment]
> 
> 
> > What should I see with those commands? @sandreenko
> 
> These are the jobs that run jit-diff to compare asm output from baseline CoreCLR with your changes.
> So it is useful to trigger them to conform that asm diffs are as expected.
> For your PR [the results](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_windows_nt_pmi_asm_diffs_prtest/24/console) are:
> 
> ```
> 11:27:16 Summary:
> 11:27:16 (Lower is better)
> 11:27:16 
> 11:27:16 Total bytes of diff: 0 (0.00 % of base)
> 11:27:16 
> 11:27:16 0 total files with size differences (0 improved, 0 regressed), 129 unchanged.
> 11:27:16 
> 11:27:16 0 total methods with size differences (0 improved, 0 regressed), 246983 unchanged.
> ```
> 
> so no diffs as expected.
> 
> [More information about jit-diff.](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_windows_nt_pmi_asm_diffs_prtest/24/console)

The second link is equals to the first one. Did you mean that?

**Comment by sandreenko** [comment]
[Fixed link](https://github.com/dotnet/jitutils/blob/master/doc/diffs.md#analyzing-diffs-jit-analyze)

File: src/jit/codegen.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (BrianBohe) [reply src/jit/codegen.h]: Thanks!

File: src/jit/codegen.h [hunk 4]
```diff
@@ -749,7 +753,7 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     void psiEndPrologScope(psiScope* scope);
 
     void psSetScopeOffset(psiScope* newScope, LclVarDsc* lclVarDsc1);
-
+#endif
```
[the remark this answers is not in the document]
Comment (BrianBohe) [reply src/jit/codegen.h]: Thanks, I have removed some useless access level/flags too.

**Comment by BrianBohe** [comment]
> \DIFF\SYSTEM.PRIVATE.CORELIB.DASM

Is true that the line is not exactly the same, but those numbers are actually the same if you compare the unwind info:

`Unwind Info:
  >> Start offset   : 0x000000 (not in unwind data)
  >>   End offset   : 0xd1ffab1e (not in unwind data)
  Version           : 1
  Flags             : 0x00
  SizeOfProlog      : 0x07
  CountOfUnwindCodes: 4
  FrameRegister     : none (0)
  FrameOffset       : N/A (no FrameRegister) (Value=0)
  UnwindCodes       :
    CodeOffset: 0x07 UnwindOp: UWOP_ALLOC_SMALL (2)     OpInfo: 3 * 8 + 8 = 32 = 0x20
    CodeOffset: 0x03 UnwindOp: UWOP_PUSH_NONVOL (0)     OpInfo: rbx (3)
    CodeOffset: 0x02 UnwindOp: UWOP_PUSH_NONVOL (0)     OpInfo: rsi (6)
    CodeOffset: 0x01 UnwindOp: UWOP_PUSH_NONVOL (0)     OpInfo: rdi (7)`

on base Line: 637596 and diff Line: 637261

**or for the other** 

`; Total bytes of code 98, prolog size 7 for method System.MemoryExtensions:AsMemory(struct):struct
; ============================================================
Unwind Info:
  >> Start offset   : 0x000000 (not in unwind data)
  >>   End offset   : 0xd1ffab1e (not in unwind data)
  Version           : 1
  Flags             : 0x00
  SizeOfProlog      : 0x07
  CountOfUnwindCodes: 4
  FrameRegister     : none (0)
  FrameOffset       : N/A (no FrameRegister) (Value=0)
  UnwindCodes       :
    CodeOffset: 0x07 UnwindOp: UWOP_ALLOC_SMALL (2)     OpInfo: 3 * 8 + 8 = 32 = 0x20
    CodeOffset: 0x03 UnwindOp: UWOP_PUSH_NONVOL (0)     OpInfo: rbx (3)
    CodeOffset: 0x02 UnwindOp: UWOP_PUSH_NONVOL (0)     OpInfo: rsi (6)
    CodeOffset: 0x01 UnwindOp: UWOP_PUSH_NONVOL (0)     OpInfo: rdi (7)
; Assembly listing for method System.MemoryExtensions:AsMemory(struct,int):struct`

on base Line: 637261, diff Line :637063

It seems just that the order in the file is different.
Should it be always the same?

**Comment by BrianBohe** [comment]
On "Checks" seems that all Azure Pipelines succeeded, but  OSX10.12 x64 Checked Innerloop Build and Test test says there is a test not passing.

Which should we see?

**Comment by BrianBohe** [comment]
@dotnet-bot Test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by BrianBohe** [comment]
I don't feel confident to assert my last two comments. If anyone could tell me whether I am right or not, you are welcome @dotnet/jit-contrib

**Comment by sandreenko** [comment]
You are right, 
> After all tests have run, I will commit changing flag to be defined by default.

Commit the change and I will merge it.

**Comment by BrianBohe** [comment]
Thanks @sandreenko

**Pull request merged**